src/typechecker.erl

```diff
@@ -275,8 +275,8 @@ compat_ty({type, _, AssocTag2, [Key2, Val2]},
              AssocTag2 == map_field_exact, AssocTag1 == map_field_exact;
              AssocTag2 == map_field_assoc, AssocTag1 == map_field_exact ->
     %% For M1 <: M2, mandatory fields in M2 must be mandatory fields in M1
-    {A1, Cs1} = compat_ty(Key1, Key2, A, TEnv),
-    {A2, Cs2} = compat_ty(Val1, Val2, A1, TEnv),
+    {A1, Cs1} = compat(Key1, Key2, A, TEnv),
+    {A2, Cs2} = compat(Val1, Val2, A1, TEnv),
```

Comment:
Good!

Reply:
There's one more case like this - `compat_ty` calls `all_type` which (probably incorrectly) calls `compat_ty` again. I have not touched that case since I don't have a test case (or even an idea for one) which could be fixed or would verify if it's intended or not. Maybe worth a separate issue?

Reply:
Good, keep the PR small.

I you want you can add a small comment such as `%% TODO: call compat/4 instead of compat_ty/4 here?`.